controller/controller.go

```diff
@@ -272,6 +349,14 @@ func deleteSchedulerHelper(logger logrus.FieldLogger, mr *models.MixedMetricsRep
 		}
 	}
 
+	err = mr.WithSegment(models.SegmentDelete, func() error {
```

Comment:
Well, the watcher should recreate it with services and pods. This might be a bug so please take some time to make sure it works as expected.